Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -91,4 +197,148 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::symbol_species_getter)
     return vm.this_value();
 }
 
+// get RegExp.input, https://github.com/tc39/proposal-regexp-legacy-features#get-regexpinput
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_getter)
+{
+    auto* regexp_constructor = vm.current_realm()->intrinsics().regexp_constructor();
+
+    // 1. Return ? GetLegacyRegExpStaticProperty(%RegExp%, this value, [[RegExpInput]]).
+    return js_string(vm, regexp_constructor->legacy_static_properties().input());
+}
+
+// get RegExp.$_, https://github.com/tc39/proposal-regexp-legacy-features#get-regexp_
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_alias_getter)
+{
+    // Keep the same implementation with `get RegExp.input`
+    return input_getter(vm);
+}
+
+// set RegExp.input, https://github.com/tc39/proposal-regexp-legacy-features#set-regexpinput--val
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_setter)
+{
+    // If numberOfArgs is 0, throw a TypeError exception.
+    if (vm.argument_count() == 0)
+        return vm.throw_completion<TypeError>(ErrorType::BadArgCountOne, "set input");
+
+    // https://github.com/tc39/proposal-regexp-legacy-features#setlegacyregexpstaticproperty-c-thisvalue-internalslotname-val-
+    // SetLegacyRegExpStaticProperty( C, thisValue, internalSlotName, val ).
+
+    // FIXME: 1. Assert C is an object that has an internal slot named internalSlotName.
+
+    // FIXME: 2. If SameValue(C, thisValue) is false, throw a TypeError exception.
+    // auto this_value = vm.this_value();
+
+    // 3. Let strVal be ? ToString(val).
+    auto val = vm.argument(0);
+    auto str_val = TRY(val.to_string(vm));
+
+    // 4. Set the value of the internal slot of C named internalSlotName to strVal.
+    auto* regexp_constructor = vm.current_realm()->intrinsics().regexp_constructor();
+    regexp_constructor->legacy_static_properties().set_input(Utf16String(str_val));
```

Comment:
You can pass the setter function as a function pointer to `SetLegacyRegExpStaticProperty`

Reply:
Only `RegExp.input` have setter, so i write the code here.

Reply:
Hmm alright, inlining functions by hand is not something we usually do, but it's probably fine here given the slightly awkward passing around of slots.